examples/bridge-app/telink/src/AppTask.cpp

```diff
@@ -0,0 +1,1094 @@
+    AddDeviceEndpoint(&gLight3, &bridgedLightEndpoint, Span<const EmberAfDeviceType>(gBridgedOnOffDeviceTypes),
+                      Span<DataVersion>(gLight3DataVersions), 1);
+
+    // Remove Light 2 -- Lights 1 & 3 will remain mapped to endpoints 3 & 5
+    RemoveDeviceEndpoint(&gLight2);
+
+    // Add Light 4 -- > will be mapped to ZCL endpoint 6
+    AddDeviceEndpoint(&gLight4, &bridgedLightEndpoint, Span<const EmberAfDeviceType>(gBridgedOnOffDeviceTypes),
+                      Span<DataVersion>(gLight4DataVersions), 1);
+
+    // Re-add Light 2 -- > will be mapped to ZCL endpoint 7
+    AddDeviceEndpoint(&gLight2, &bridgedLightEndpoint, Span<const EmberAfDeviceType>(gBridgedOnOffDeviceTypes),
+                      Span<DataVersion>(gLight2DataVersions), 1);
+
+    // Add Temperature Sensor devices --> will be mapped to endpoint 8
+    AddDeviceEndpoint(&TempSensor1, &bridgedTempSensorEndpoint, Span<const EmberAfDeviceType>(gBridgedTempSensorDeviceTypes),
+                      Span<DataVersion>(gTempSensor1DataVersions), 1);
+}
+
+void HandleDeviceTempSensorStatusChanged(DeviceTempSensor * dev, DeviceTempSensor::Changed_t itemChangedMask)
+{
+    using namespace Clusters;
+    if (itemChangedMask &
+        (DeviceTempSensor::kChanged_Reachable | DeviceTempSensor::kChanged_Name | DeviceTempSensor::kChanged_Location))
+    {
+        HandleDeviceStatusChanged(static_cast<Device *>(dev), (Device::Changed_t) itemChangedMask);
+    }
+    if (itemChangedMask & DeviceTempSensor::kChanged_MeasurementValue)
+    {
+        ScheduleReportingCallback(dev, TemperatureMeasurement::Id, TemperatureMeasurement::Attributes::MeasuredValue::Id);
+    }
+}
+
+EmberAfStatus HandleReadTempMeasurementAttribute(DeviceTempSensor * dev, chip::AttributeId attributeId, uint8_t * buffer,
+                                                 uint16_t maxReadLength)
+{
+    using namespace Clusters::TemperatureMeasurement::Attributes;
+
+    if ((attributeId == MeasuredValue::Id) && (maxReadLength == 2))
+    {
+        int16_t measuredValue = dev->GetMeasuredValue();
+        memcpy(buffer, &measuredValue, sizeof(measuredValue));
+    }
+    else if ((attributeId == MinMeasuredValue::Id) && (maxReadLength == 2))
+    {
+        int16_t minValue = dev->mMin;
+        memcpy(buffer, &minValue, sizeof(minValue));
+    }
+    else if ((attributeId == MaxMeasuredValue::Id) && (maxReadLength == 2))
+    {
+        int16_t maxValue = dev->mMax;
+        memcpy(buffer, &maxValue, sizeof(maxValue));
+    }
+    else if ((attributeId == FeatureMap::Id) && (maxReadLength == 4))
+    {
+        uint32_t featureMap;
+        FeatureMap::Get(dev->GetEndpointId(), &featureMap);
+        memcpy(buffer, &featureMap, sizeof(featureMap));
+    }
+    else if ((attributeId == ClusterRevision::Id) && (maxReadLength == 2))
+    {
+        uint16_t clusterRevision;
+        ClusterRevision::Get(dev->GetEndpointId(), &clusterRevision);
```

Comment:
This is just going to call right back into this function, no?  Was this tested?

Reply:
It's been falling in infinite loop indeed, fixed it already, thanks. Tested manually with chip-tool, all 3 clusters have now correct respond on `read cluster-revision` command 
> ./chip-tool onoff read cluster-revision {NODE_ID} {ENDPOINT_ID}

> 
>  			AttributeDataIB =
>  			{
>  				DataVersion = 0x6c62e585,
> 				AttributePathIB =
> 				{
>  					Endpoint = 0x6,
>  					Cluster = 0x6,
>  					Attribute = 0x0000_FFFD,
>  				}
>  					
>  				Data = 4, 
> 			},